spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListener.java

```diff
@@ -241,4 +241,13 @@
 	 */
 	String[] properties() default {};
 
+	/**
+	 * When set to true, enable the listener code to access the consumer's {@link code
+	 * group.id} property using a {@code @Header(KafkaHeaders.GROUP_ID} String groupId)
+	 * method parameter. Useful if you use the same listener code in multiple containers.
+	 * @return true to provide the header.
+	 * @since 2.3
+	 */
+	String exposeGroupId() default "false";
```

Comment:
I wonder why would one doesn't want to map this property to header?..
Why don't we do that unconditionally like for many other Kafka records properties?

Reply:
Doh - good point 😄